internal/translation/deployment/deployment.go

```diff
@@ -132,16 +157,37 @@ func (ds *ProductionAtlasDeployments) DeploymentIsReady(ctx context.Context, pro
 	return clusterStatus.GetChangeStatus() == string(mongodbatlas.ChangeStatusApplied), nil
 }
 
-func (ds *ProductionAtlasDeployments) GetDeployment(ctx context.Context, projectID, name string) (Deployment, error) {
+func (ds *ProductionAtlasDeployments) GetFlexCluster(ctx context.Context, projectID, name string) (*Flex, error) {
+	if ds.isGov {
+		return nil, nil
```

Comment:
Isn't flex supported by gov? in that case I suggest also add a constraint to our support checker https://github.com/mongodb/mongodb-atlas-kubernetes/blob/0f1f8007e8b4f514b468ad236d168d00fba608fb/internal/controller/atlas/provider.go#L78-L106